### ISSUE TITLE
getting the outputs for a "wallet send" appears to be slow

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin#246]
For `wallet send` we see the log msgs scrolling past slowly - 

```
Nov 09 04:08:30.726 TRCE Derived Key key_id: 81914098603b2a70dd39
Nov 09 04:08:31.707 TRCE Derived Key key_id: 81914098603b2a70dd39
Nov 09 04:08:32.278 TRCE Derived Key key_id: 3ba8170e9f3a02af2437
Nov 09 04:08:32.856 TRCE Derived Key key_id: c436d51e17cf79eefcdd
Nov 09 04:08:33.679 TRCE Derived Key key_id: bfd9c0532ee55ca79b7d
Nov 09 04:08:34.256 TRCE Derived Key key_id: bfd9c0532ee55ca79b7d
```

For `wallet info` these same log msgs are really fast - 

```
Nov 09 04:09:31.969 TRCE Derived Key key_id: bfd9c0532ee55ca79b7d
Nov 09 04:09:31.970 TRCE Derived Key key_id: 3ba8170e9f3a02af2437
Nov 09 04:09:32.048 TRCE Derived Key key_id: c436d51e17cf79eefcdd
Nov 09 04:09:32.048 TRCE Derived Key key_id: 81914098603b2a70dd39
Nov 09 04:09:32.049 TRCE Derived Key key_id: 805e7a4d3d3966bb96b8
Nov 09 04:09:32.049 TRCE Derived Key key_id: 9dad3e0b80566a19a3bc
```

Also - why are there duplicates in those log msgs for `wallet send` (those keys/identifiers should be unique)?

**Comment by antiochp** [comment]
Should be resolved by #253 (cache key derivations by key_id).

**Comment by antiochp** [comment]
Resolving.
